app/Http/Controllers/RestApi/RestApiMember.php

```diff
@@ -2,62 +2,95 @@
-use Symfony\Component\Yaml\Exception\ParseException;
-use Symfony\Component\Yaml\Yaml;
-use App\Exceptions\IllegalArgumentException;
-
-use App\Http\Controllers\RestApi;
-
-use Illuminate\Http\Request;
-
 /**
  * Class RestApiMember
  *
  * Manages all API resources connected to the Member
  */
-class RestApiMember
-{
-
-  /**
-  * Return a json with the member fields
-  *
-  * @param $request - the http Request
-  * @param $member_id - the id of the member that we should get
-  * @param $is_admin - is the call by an admin resource (i.e. should we return
-  *                     all information about the member)
-  * @return string  the JSON
-  */
-  public function getMember($request, $member_id, $is_admin = false) {
-    ApiHelper::checkIntegerInput($member_id);
-    $memberRepo = ApiHelper::createMemberRepo($request->header($key = 'db_key'));
-
-    $member = $memberRepo->get($member_id);
-
-    $data = ApiHelper::getMemberAsArray($member, $is_admin);
-
-    return json_encode($data);
-  }
-
-  /**
-  * Return a json with the all member that changed since the $revisionId
-  *
-  * @return string the JSON
-  */
-  public function getChanged($request, $revisionId) {
-    ApiHelper::checkIntegerInput($revisionId);
-    $memberRepo = ApiHelper::createMemberRepo($request->header($key = 'db_key'));
-
-    $members = $memberRepo->getUpdated($revisionId);
-
-    $data = [];
-    foreach ($members as $member) {
-      $data[$member->id] = ApiHelper::getMemberAsArray($member);
-    }
-
-    return json_encode($data);
-  }
+class RestApiMember {
+
+	/**
+	 * Return a json with the member fields
+	 *
+	 * @param $request - the http Request
+	 * @param $member_id - the id of the member that we should get
+	 * @param $is_admin - is the call by an admin resource (i.e. should we return
+	 *                     all information about the member)
+	 *
+	 * @return string  the JSON
+	 */
+	public function getMember( $request, $member_id, $is_admin = false ) {
+		ApiHelper::checkIntegerInput( $member_id );
+		$memberRepo = ApiHelper::createMemberRepo( $request->header( $key = 'db_key' ) );
+
+		$member = $memberRepo->get( $member_id );
+
+		$data = ApiHelper::getMemberAsArray( $member, $is_admin );
+
+		return json_encode( $data );
+	}
+
+	/**
+	 * Return a json with the member fields of the main record (!) determined
+	 * by {@link MemberRepository::getMaster()}.
+	 *
+	 * @param $request - the http Request
+	 * @param $member_id - the id of the member that we should get
+	 * @param $group_ids - string with commaseparated list of ids of root group
+	 *                     below which the main record should be searched
+	 *                     {@see MemberRepository::getMaster()}
+	 * @param $is_admin - is the call by an admin resource (i.e. should we return
+	 *                     all information about the member)
+	 *
+	 * @return string  the JSON
+	 */
+	public function getMainMember( $request, $member_id, $group_ids, $is_admin = false ) {
```

Comment:
Woher kommt der Wert in $group_ids jeweils, dass Du ihn als String übergibst? Kommt der immer aus einer Text-Ressource, z.B. Config-File oder kann es auch vorkommen, dass er irgendwo im Code definiert wird? Falls letzteres der Fall ist, würde ich ein Array übergeben, wenn nicht ist die explode Variante wohl ok (es kommt mir zwar unorthodox vor, aber die Gründe dafür haben in PHP keine Gültigkeit)

Reply:
die `$group_ids` kommt aus der url `/api/v1/member/{member_id}/main/{search_groups_ids}` darum als kommaseparierter String. Vermutlich wird es in 99% der Fälle nur eine id sein, aber mit der explode Option ist mit sehr wenig Aufwand viel Felxibilität gewonnen.